src/viam/sdk/rpc/dial.hpp

```diff
@@ -16,9 +16,15 @@ class ViamChannel {
    public:
     void close();
     ViamChannel(std::shared_ptr<GrpcChannel> channel, const char* path, void* runtime);
+
     static std::shared_ptr<ViamChannel> dial(const char* uri,
                                              const boost::optional<DialOptions>& options);
 
+    // Specifies that this dial is an initial connection attempt and that initial connection options
+    // should be used.
+    static std::shared_ptr<ViamChannel> dial_initial(const char* uri,
```

Comment:
I'm a little curious how one is to use `dial` and `dial_initial` now. If I'm writing a loop to connect, does it look like this:

```
bool first = false;
while true {
    if (first) {
        first = false;
        if (auto chan = dial_initial(...); chan)
            return chan;
    } else {
        if (auto chan = dial(...); chan)
            return chan;
    }
}
```

It feels a little awkward for the user.

Reply:
I think in most cases users shouldn't use `dial_initial` at all. The preferred entrypoint for creating a robot client is `RobotClient::at_address(...)` which calls `dial` itself, so the user doesn't have to know about either.

If a user _does_ need to create a channel themselves for whatever reason, then they would just call `dial_initial(...)` if they want to specify an initial timeout at that time (say they're trying to dial to a robot as it's starting up so they want to retry), or `dial(...)` otherwise. The `while true` loop above is redundant; `dial_initial(...)` is a wrapper around calls to `dial` already.

Reply:
Just for explicitness: if you want something akin to this loop where you try to connect forever until it succeeds, you would do the following:

```
DialOptions opts;
opts.set_initial_connection_attempts(0); // 0 is a sentinel value that says "try forever"
return ViamChannel::dial_initial(uri, opts);
```

Reply:
Got it, thanks for clarifying.